policy-controller/k8s/status/src/http_route.rs

```diff
@@ -19,39 +19,224 @@ pub enum ParentReference {
     let namespace = http_route
         .metadata
         .namespace
+        .as_deref()
         .expect("HTTPRoute must have a namespace");
     http_route
         .spec
         .inner
         .parent_refs
-        .into_iter()
+        .iter()
+        .flatten()
+        .map(|parent_ref| ParentReference::from_parent_ref(parent_ref, namespace))
+        .collect()
+}
+
+pub(crate) fn make_backends(http_route: &policy::HttpRoute) -> Vec<BackendReference> {
+    let namespace = http_route
+        .metadata
+        .namespace
+        .as_deref()
+        .expect("HTTPRoute must have a namespace");
+    http_route
+        .spec
+        .rules
+        .iter()
         .flatten()
-        .map(|parent_ref| ParentReference::from_parent_ref(parent_ref, &namespace))
+        .flat_map(|rule| rule.backend_refs.iter().flatten())
+        .filter_map(|http_backend_ref| http_backend_ref.backend_ref.as_ref())
+        .map(|backend_ref| BackendReference::from_backend_ref(&backend_ref.inner, namespace))
         .collect()
 }
 
 impl ParentReference {
-    fn from_parent_ref(parent_ref: gateway::ParentReference, default_namespace: &str) -> Self {
-        if policy::httproute::parent_ref_targets_kind::<Server>(&parent_ref) {
+    fn from_parent_ref(parent_ref: &gateway::ParentReference, default_namespace: &str) -> Self {
+        if policy::httproute::parent_ref_targets_kind::<Server>(parent_ref) {
             // If the parent reference does not have a namespace, default to using
             // the HTTPRoute's namespace.
-            let namespace = parent_ref
-                .namespace
-                .unwrap_or_else(|| default_namespace.to_string());
-            ParentReference::Server(ResourceId::new(namespace, parent_ref.name))
-        } else if policy::httproute::parent_ref_targets_kind::<Service>(&parent_ref) {
+            let namespace = parent_ref.namespace.as_deref().unwrap_or(default_namespace);
+            ParentReference::Server(ResourceId::new(
+                namespace.to_string(),
+                parent_ref.name.clone(),
+            ))
+        } else if policy::httproute::parent_ref_targets_kind::<Service>(parent_ref) {
             // If the parent reference does not have a namespace, default to using
             // the HTTPRoute's namespace.
-            let namespace = parent_ref
-                .namespace
-                .unwrap_or_else(|| default_namespace.to_string());
-            ParentReference::Service(ResourceId::new(namespace, parent_ref.name), parent_ref.port)
+            let namespace = parent_ref.namespace.as_deref().unwrap_or(default_namespace);
+            ParentReference::Service(
+                ResourceId::new(namespace.to_string(), parent_ref.name.clone()),
+                parent_ref.port,
+            )
         } else {
             ParentReference::UnknownKind
         }
     }
 }
+
+impl BackendReference {
+    fn from_backend_ref(
```

Comment:
It occurs to me that this may be consistent with what we do elsewhere, which would be a good reason to keep it as-is. Still... maybe a pattern we want to revisit in the future.

Reply:
Yeah, it's not super intuitive. I considered maybe doing a `From<>` impl but that complicated things since we needed to take a namespace default. Could've implemented it on a tuple it just made everything more complicated than it had to be. `new` would be a good choice... will leave as is for now but will keep it in mind.